tests/test_util.py

```diff
@@ -31,3 +31,26 @@ async def test_ensure_directory_absent(tmp_path: pathlib.Path):
 
     await ensure_absent(path)
     assert not await aiofiles.os.path.exists(path)
+
+
+@pytest.mark.asyncio
+async def test_ensure_absent_on_directory_without_write_permissions(
+    tmp_path: pathlib.Path,
+):
+    # Create a directory
+    path = tmp_path / "test_dir"
+    await aiofiles.os.mkdir(path)
+
+    # Change permissions to remove write access (read and execute only)
+    os.chmod(path, 0o500)
+
+    # Try to remove the directory - should raise a PermissionError or OSError
+    with pytest.raises((PermissionError, OSError)):
```

Comment:
This will not work if the tests are executed as root, root can do anything to a directory.